tests/pytest/oauth/test_views.py

```diff
@@ -80,17 +108,44 @@ def test_logout(mocker, app_request):
     assert session.enrollment_token(app_request) is False
 
 
+def test_logout_analytics(mocker, app_request):
+    spy = mocker.patch("benefits.oauth.analytics.started_sign_out")
+
+    # logout internally calls _deauthorize_redirect
+    # this mocks that function and a success response
+    message = "logout successful"
+    mocker.patch("benefits.oauth.views._deauthorize_redirect", return_value=HttpResponse(message))
+
+    token = "token"
+    session.update(app_request, oauth_token=token)
+    assert session.oauth_token(app_request) == token
+
+    logout(app_request)
+
+    spy.assert_called_once()
+
+
 @pytest.mark.request_path("/oauth/post_logout")
-def test_post_logout(app_request):
+def test_post_logout(mocker, app_request):
     origin = reverse("core:index")
     session.update(app_request, origin=origin)
 
+    mocker.patch.object(benefits.oauth.views, "analytics")
```

Comment:
This doesn't seem to be necessary if the test function is decorated with `@pytest.mark.django_db`. I'd go with that to be more consistent with other tests, test more of the code, and avoid crowding the test body.

Reply:
I agree that that's an alternative. I would prefer though to use `@pytest.mark.django_db` only when the test is for functionality that actually requires the database. 

These tests, e.g. `test_logout`, all of a sudden "want" to use the database only because sending an analytic event requires reading from the session which requires trying to query for models like `TransitAgency` and `EligibilityVerifier`. (The test actually doesn't even set up an agency or verifier to be read from the session, so we'd be querying the database just to get back nothing.) Since the intention of the test is to cover login/logout rather than analytics, I think mocking analytics makes that intention clearer.

All that said, I do wish I could use a decorator or something to mock it instead of crowding the test body.

Reply:
> All that said, I do wish I could use a decorator or something to mock it instead of crowding the test body.

I think we should be able to do this with a fixture/mark combo?

Similar to whatever the `@pytest.mark.django_db` is doing to signal that we "need the db" - can we have something like `@pytest.mark.mock_analytics`? Not necessary for this PR I think.

Reply:
Created #503